src/transformers/models/gpt_neox/tokenization_gpt_neox.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+# Copyright 2022 EleutherAI and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for GPTNeoX."""
+from ...utils import logging
+from ..gpt2.tokenization_gpt2 import GPT2Tokenizer
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.json", "merges_file": "merges.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "gpt-neox-20b": "https://huggingface.co/gpt-neox-20b/resolve/main/vocab.json",
+    },
+    "merges_file": {
+        "gpt-neox-20b": "https://huggingface.co/gpt-neox-20b/resolve/main/merges.txt",
+    },
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "gpt-neox-20b": 1024,
+}
+
+
+class GPTNeoXTokenizer(GPT2Tokenizer):
```

Comment:
In that situation, if the two are equivalent, then there is no need to create a new tokenizer. It's supported to have a mismatch between tokenizer and model!

For example, that's what GPTNeo does. See the `tokenizer_class` attribute of the configuration: https://huggingface.co/EleutherAI/gpt-neo-1.3B/blob/main/config.json

We encourage you to add it to the `tokenizer_config.json` file instead so that the tokenizer may be loaded even when the model config file isn't present, but both are supported.

You'll need to update the `tokenization_auto` file to map correctly as well, see for example for GPTNeo:

https://github.com/huggingface/transformers/blob/c79bbc3ba54a81dab2eac13d89f264ed64cb2460/src/transformers/models/auto/tokenization_auto.py#L182